it/track_repo_compatibility/conftest.py

```diff
@@ -0,0 +1,67 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import shlex
+import subprocess
+
+import pytest
+
+RALLY_HOME = os.getenv("RALLY_HOME", os.path.expanduser("~"))
+RALLY_CONFIG_DIR = os.path.join(RALLY_HOME, ".rally")
+RALLY_TRACKS_DIR = os.path.join(RALLY_CONFIG_DIR, "benchmarks", "tracks", "default")
```

Comment:
Should we consider another location in other to not affect and be affected by changes and checkouts in the default tracks git repository?

Reply:
I was originally intending to have these tests mimic a clean installation of Rally. By default, if a user installs Rally, it will clone `rally-tracks` into `~/.rally/benchmarks/tracks/default` the first time an `esrally` command is run that requires a track repository to be available on disk.

I think this makes sense for CI, but I can see the argument for instead explicitly cloning into a distinct, non-default location for these tests, especially locally.

[This commit ](https://github.com/elastic/rally/pull/1564/commits/257794baaf8cb4228196cc63c8ff7172bf2da86e)implements the following control flow:

- If `--track-repository` is provided directly, we make sure that the directory actually exists on disk, raising an exception if it doesn't.
- If `--track-repository` isn't provided, we default to `~/.rally/benchmarks/tracks/rally-tracks-compat`, cloning `https://github.com/elastic/rally-tracks` there if the directory doesn't yet exist.

Re: your suggestions [here](https://github.com/elastic/rally/pull/1564#discussion_r953452173), shelling out to `git` does seem easier for cloning, so I went with that.

I considered making other things configurable (the remote repo URL, an option to perform an initial clone into a custom track repository path, etc.) but I think this approach serves CI and local development well enough in their default cases.

What do you think?

Reply:
On one hand, I don't want to conflict with people's local configurations. On the other hand, I don't want additional checkouts present on my local in order to isolate testing properly.

I propose creating a `rally-compat.ini` which has a comment at the header `# Automatically managed - do not touch` and running compatibility tests via `--configuration-name compat`. This would specify the canonical upstream `default.url = https://github.com/elastic/rally-tracks` (as well as our `teams` and `source` URLs) in a place that doesn't need to be touched by users and would better avoid users shooting themselves in the foot. Is this feasible?

Reply:
Mike and I discussed this offline. This is not a huge concern compared to the trappy nature of trying to isolate it perfectly. I agree with him that we could just move forward with this approach as-is

Reply:
Yeah I also think the rally-tracks-compat checkout is the best compromise here. Thanks!